pkg/testutils/testcluster/testcluster.go

```diff
@@ -301,19 +303,54 @@ func (tc *TestCluster) Start(t testing.TB) {
 		errCh = make(chan error, nodes)
 	}
 
+	// Determine if we should probabilistically start a SQL server for the
+	// cluster.
+	const probabilityOfStartingTestTenant = 0.5
+	probabilisticallyStartTestSQLServer := true
+	startedTestSQLServer := true
+	if rand.Float32() > probabilityOfStartingTestTenant {
+		probabilisticallyStartTestSQLServer = false
+	}
+
 	disableLBS := false
 	for i := 0; i < nodes; i++ {
 		// Disable LBS if any server has a very low scan interval.
 		if tc.serverArgs[i].ScanInterval > 0 && tc.serverArgs[i].ScanInterval <= 100*time.Millisecond {
 			disableLBS = true
 		}
 
+		// If we're not probabilistically starting the test SQL server, disable
+		// its start and set the "started" flag accordingly. We need to do this
+		// with two separate if checks because the DisableDefaultSQLServer flag
+		// could have been set coming into this function by the caller.
+		if !probabilisticallyStartTestSQLServer {
+			tc.Servers[i].Cfg.DisableDefaultSQLServer = true
+		}
+		if tc.Servers[i].Cfg.DisableDefaultSQLServer {
+			if startedTestSQLServer && i > 0 {
+				t.Fatal(errors.Newf("starting only some nodes with a test SQL server is not"+
+					"currently supported - attempted to disable SQL sever on node %d", i))
+			}
+			startedTestSQLServer = false
+		}
+
 		if tc.clusterArgs.ParallelStart {
 			go func(i int) {
 				errCh <- tc.startServer(i, tc.serverArgs[i])
 			}(i)
 		} else {
 			if err := tc.startServer(i, tc.serverArgs[i]); err != nil {
+				if strings.Contains(err.Error(), "requires a CCL binary") {
```

Comment:
Can we put a sentinel error somewhere and use `errors.Is` instead?
Doesn't have to be this PR.